### DISCUSSION TITLE
How to import modules within plugins and avoid MODULE_NOT_FOUND from the webpack-dev-middleware

### DISCUSSION THREAD

**Discussion by TomStrepsil** [webpack/webpack#18936] [Q&A]
I'm writing a webpack plugin that imports a module, based on a file path, similar to:
```js
const config = (await import(path)).default();
```
This isn't a file that is otherwise imported into the compilation as a module.

This works fine during build, and using HMR without `webpack-dev-middleware`

However, despite the `inputFileSystem` recognising the file path, `MODULE_NOT_FOUND` is thrown from the `webpack-dev-middleware` compile, when attempting the above `import`.

I've tried adding a file dependency to the compilation, to various modules at various places during build, creating loaders that add a dependency, all to no avail.

Has anyone got experience of importing a module from disk during a compilation and having `webpack-dev-middleware` not complain?

**Comment by alexander-akait** [comment]
@asos-tomp Can you show the source code of the plugin (even better with a code example of how it reproduce) so that I can study it and why this happens in your case?
